r/R/dataset.R

```diff
@@ -25,21 +25,28 @@
 #'    parsed, and path segments will be matched with the schema fields. For
 #'    example, `schema(year = int16(), month = int8())` would create partitions
 #'    for file paths like "2019/01/file.parquet", "2019/02/file.parquet", etc.
+#'   * A character vector that defines the field names corresponding to those
+#'    path segments (that is, you're providing the names that would correspond
+#'    to a `Schema` but the types will be autodetected)
 #'   * A `HivePartitionScheme`, as returned by [hive_partition()]
```

Comment:
This is slightly confusing since `hive_partition()` (with no arguments) now returns a Discovery instead of a Scheme
```suggestion
#'   * A `HivePartitionScheme`, as returned by [hive_partition(...)]
```

Reply:
It's somewhat academic since neither `HivePartitionScheme` nor `HivePartitionSchemeDiscovery` are useful to a user except as an input here, but if we wanted this to be completely factual, I'd prefer this: 

```suggestion
#'   * A `HivePartitionScheme` or `HivePartitionSchemeDiscovery`, as returned by [hive_partition()]
```

The docs for `hive_partition()` do explain that it could return either: https://github.com/apache/arrow/pull/6077/files#diff-b3408619fab15d6abbec82111f21bb8cR13

Reply:
perfect